core/src/main/java/org/apache/brooklyn/enricher/stock/reducer/StringStringReducer.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.enricher.stock.reducer;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import com.google.common.base.Function;
+
+public class StringStringReducer extends GenericStringReducer<String> {
+    
+    public StringStringReducer() {}
+
+    @Override
+    protected Function<List<String>, String> createReducerFunction(
+            String reducerName, Map<String, ?> parameters) {
+        Function<List<String>, String> function = super.createReducerFunction(reducerName, parameters);
+        if(function != null) return function;
+        
+        if(Objects.equals(reducerName, "joiner")){
```

Comment:
This feels like a strange place to define the names of functions that we'll accept.

I presume these conveniences are for yaml (as in Java one would just use `StringFunctions.joiner()` directly)?

Should we be adding support for different functions to https://github.com/nakomis/incubator-brooklyn/blob/master/usage/camp/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/methods/BrooklynDslCommon.java ? As is done in https://github.com/apache/incubator-brooklyn/pull/910/files#diff-d108c16ec590974e317590c54f71e9da